ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -845,6 +852,91 @@ namespace OrdinalAxis {
+         *
+         * @param {Array<number>} [ordinalPositions]
+         *        An array of points visible in the current plot area.
+         *
+         * @param {Array<number>} [extendedOrdinalPositions]
+         *        An array of all points available on the axis
+         *        for the given data set.
+         *
+         * @return {number}
+         */
+        axisProto.getIndexOfPoint = function (
+            val: number,
+            ordinalPositions: Array<number>,
+            extendedOrdinalPositions: Array<number>
+        ): number {
+            const axis = this,
+                // Distance in pixels between two points
+                // on the ordinal axis in the current zoom.
+                ordinalPointPixelInterval = axis.len / ordinalPositions.length - 1,
+                shiftIndex = val / ordinalPointPixelInterval;
+
+            // Make sure that the returned index does not exceed array
+            // boundaries. If so, return the first or the last index of array.
+            return Math.min(
+                Math.max(
+                    extendedOrdinalPositions.indexOf(axis.min as any) + shiftIndex,
+                    0
+                ), // first index
+                extendedOrdinalPositions.length - 1 // last index
+            );
+        };
+
+        /**
+         * Get axis position of given index of the extended ordinal positions.
+         * Used only when panning an ordinal axis.
+         *
+         * @private
+         * @function Highcharts.Axis#lin2val
+         *
+         * @param {number} index
+         *        The index value of searched point
+         *
+         * @return {number}
+         */
+        axisProto.index2val = function (index: number): number {
+            let axis = this,
+                ordinal = axis.ordinal,
+                // Context could be changed to extendedOrdinalPositions.
+                ordinalPositions = ordinal.positions;
+
+            // The visible range contains only equally spaced values.
+            if (!ordinalPositions) {
+                return index; // TODO Find better return value if ordinal doesn't exist.
+            }
+
+            let i = ordinalPositions.length - 1,
+                distance;
+
+            if (index < 0) { // out of range, in effect panning to the left
+                index = ordinalPositions[0];
+            } else if (index > i) { // out of range, panning to the right
+                index = ordinalPositions[i];
+            } else { // split it up
+                i = Math.floor(index);
+                distance = index - i; // the decimal
+            }
+
+            if (typeof distance !== 'undefined' &&
+                typeof ordinalPositions[i] !== 'undefined') {
+                return ordinalPositions[i] + (distance ?
+                    distance *
+                        (ordinalPositions[i + 1] - ordinalPositions[i]) :
+                    0);
+            }
+            return index;
```

Comment:
is it even possible to get this return value?

Reply:
From the TS point of view, it is, so I think it should stay here.